module/zfs/arc.c

```diff
@@ -10145,16 +10138,22 @@ l2arc_log_blk_read(l2arc_dev_t *dev,
 	switch (L2BLK_GET_COMPRESS((this_lbp)->lbp_prop)) {
 	case ZIO_COMPRESS_OFF:
 		break;
-	case ZIO_COMPRESS_LZ4:
-		abd = abd_alloc_for_io(asize, B_TRUE);
+	case ZIO_COMPRESS_LZ4: {
+		abd_t *abd = abd_alloc_for_io(asize, B_TRUE);
```

Comment:
We could use `abd_alloc_linear()` here, since it is transient and never actually goes through I/O.

Reply:
And same!